File: csharp-example/csharp-example/pages/AdminPanelLoginPage.cs
using System;
using OpenQA.Selenium;

namespace csharp_example
{
    internal class AdminPanelLoginPage : Page
    {
        public AdminPanelLoginPage(IWebDriver driver) : base(driver) { }

        internal AdminPanelLoginPage Open()
        {
            driver.Url = "http://localhost/litecart/admin";
            return this;
        }

        internal bool IsOnThisPage()
        {
            return driver.FindElements(By.Id("box-login")).Count > 0;
        }

        internal AdminPanelLoginPage EnterUsername(string username)
        {
            driver.FindElement(By.Name("username")).SendKeys(username);
            return this;
        }

        internal AdminPanelLoginPage EnterPassword(string password)
        {
            driver.FindElement(By.Name("password")).SendKeys(password);
            return this;
        }

        internal void SubmitLogin()
        {
            driver.FindElement(By.Name("login")).Click();
            wait.Until(d => d.FindElement(By.Id("box-apps-menu")));
        }

    }
}

File: js-example/pages/admin_panel_login_page.js
var webdriver = require('selenium-webdriver');

var By = webdriver.By,
    until = webdriver.until;

class AdminPanelLoginPage {

    constructor(driver) {
        this.driver = driver;
    }

    open() {
        this.driver.get("http://localhost/litecart/admin");
        return this;
    }

    onThisPageDo(f) {
        this.driver.findElements(By.id("box-login")).then(function(arr) {
            if (arr.length > 0) {
                f.call();
            }
        });
    }

    enterUsername(username) {
        this.driver.findElement(By.name("username")).sendKeys(username);
        return this;
    }

    enterPassword(password) {
        this.driver.findElement(By.name("password")).sendKeys(password);
        return this;
    }

    submitLogin() {
        this.driver.findElement(By.name("login")).click();
        this.driver.wait(until.elementLocated(By.id("box-apps-menu")));
    }

}

exports.AdminPanelLoginPage = AdminPanelLoginPage;
